compiler/pipes/gen-tree-postprocess.cpp

```diff
@@ -230,9 +230,9 @@ VertexPtr GenTreePostprocessPass::on_enter_vertex(VertexPtr root) {
 }
 
 VertexPtr GenTreePostprocessPass::on_exit_vertex(VertexPtr root) {
-  if (root->type() == op_var) {
-    if (VertexUtil::is_superglobal(root->get_string())) {
-      root->extra_type = op_ex_var_superglobal;
+  if (auto as_var = root.try_as<op_var>()) {
+    if (as_var->str_val[0] == '_' && VarData::does_name_eq_any_language_superglobal(as_var->str_val)) {
```

Comment:
```suggestion
    if (VarData::does_name_eq_any_language_superglobal(as_var->str_val)) {
```

Reply:
Though this suggestion is correct, it would affect performance: all $variables would be checked, but we have a guarantee, that superglobals start from _